src/server/server.js

```diff
@@ -38,6 +38,16 @@ app.get("/stock/:nse", (req, res) => {
   });
 });
 
+app.get("/stockChart/:params", (req, res) => {
+  const {params} = encodeURIComponent(req.params);
```

Comment:
This looks suspect to me. encodeURIComponent expects a string, but I'm pretty sure req.params is not a string, it's a map (param_name=>param) if I'm looking at this right. The function you want would be https://nodejs.org/api/querystring.html#querystring_querystring_stringify_obj_sep_eq_options (querystring.stringify). My recommendation would be to do the querystring.stringify in the getStockChartUrl method and pass your parameters as a map to that function.